src/Reflection/ClassReflection.php

```
@@ -945,7 +946,11 @@ public function getEnumCases(): array
 		foreach ($this->reflection->getCases() as $case) {
 			$valueType = null;
 			if ($case instanceof ReflectionEnumBackedCase) {
-				$valueType = $this->initializerExprTypeResolver->getType($case->getValueExpression(), $initializerExprContext);
+				try {
+					$valueType = $this->initializerExprTypeResolver->getType($case->getValueExpression(), $initializerExprContext);
+				} catch (LogicException) {
```

Catch and swallow of such a generic exception could backfire easily in the future. 

Can't we detect the situation somehow? Maybe by using the Reflection class ?

---

I thought about it.

I hesitated about introducing a hasValue method in the betterException class. But I dunno if ondrej wants to add extra method in his fork

What I can do is to just catch the needed call